### ISSUE TITLE
MRD 

### ISSUE THREAD

**Issue by duanbing** [openinx/huker#24]
Aim at  deploying Hadoop components  efficiently,  with built-in monitoring and alerting service and page view.  
   firstly, we should abstract the deployment procedure, we know a complete requests flow looks like a DAG(directed acyclic graph) , so all the compononts is the point in the graph, and the backend IP will be assumed to be edges. 
   secondly, we need abstract the specification of single componets, like hdfs node or zookeeper server. docker is the first thing we could use.  for monitoring,  we install the collecting  agent , configuration updating agent in the docker container.

**Comment by openinx** [comment]
> Aim at deploying Hadoop components efficiently, with built-in monitoring and alerting service and page view.
Yes,  IMO , Hadoop deployment/monitor/alter is the most important parts for Huker.   

DAG is a nice design, but currently, for huker it's too complex.   The deployment feature is not stable yet, so too early to implement it now.   Will think about it  after its first release.

The same for docker & k8s. 

Thanks @duanbing
